### PR TITLE
Allow the sentry_key to update the existing resource based on key_id

### PR DESCRIPTION
**Pull request by fa93hws**

### CODE REVIEW

File: sentry/resource_sentry_key.go [hunk 1]
```diff
@@ -74,6 +80,11 @@ func resourceSentryKey() *schema.Resource {
 }
 
 func resourceSentryKeyCreate(d *schema.ResourceData, meta interface{}) error {
+	inputKeyId := d.Get("key_id").(string)
+	if inputKeyId != "" {
+		d.SetId(inputKeyId)
+		return resourceSentryKeyUpdate(d, meta)
```
Comment (olebedev) [inline sentry/resource_sentry_key.go:86]: I little bit concerned about update action which is being applied while creating a key. Can we move this condition on the level upper? Something like:

```golang
    inputKeyId := d.Get("key_id").(string)
    if inputKeyId != "" {
        d.SetId(inputKeyId)
        return resourceSentryKeyUpdate(d, meta)
    } else {
        return resourceSentryKeyCreate(d, meta)
    }

```

**Review by olebedev — COMMENTED** [review]
Could you please give more context on what's going on here, links to docs, etc?

**Comment by fa93hws** [comment]
> Could you please give more context on what's going on here, links to docs, etc?

The intent of this PR is to solve the issue that the plugin can not update a resource based on the key id.
`terraform import` can not be used here because the key will be generated automatically when a Sentry project is created. Doing `terraform import` every time for the project creation is not a good user experience.

This PR add a `key_id` field to allow the updating based on the given key.

terraform plugin docs: https://www.terraform.io/docs/plugins/provider.html (I am not familiar with it either😅)

**Comment by fa93hws** [comment]
PR closed in favor of https://canva.slack.com/archives/CB5R5LJ10/p1573437132005100

**Pull request closed without merging**